sympy/matrices/expressions/matexpr.py

```diff
@@ -171,8 +171,15 @@ def cols(self):
         return self.shape[1]
 
     @property
-    def is_square(self):
-        return self.rows == self.cols
+    def is_square(self) -> Optional[bool]:
+        try:
+            diff = self.rows - self.cols
+        except TypeError:
+            # XXX This error is raised because of the existance of GenericZeroMatrix, GenericIdentity
+            return None
+        if isinstance(diff, Basic):
+            return diff.is_zero
+        return diff == 0
```

Comment:
This all seems a bit unfortunate. Ideally there would be consistent types here. I wonder if maybe the generic matrices should return some dummy symbol as well. At least for GenericIdentity it should be possible to say that `is_square` is true but maybe this is just something to be handled in those subclasses rather than here.

Reply:
This is avoided by adding `is_square` check to GenericIdentity. I'm not sure about the consequences of using dummy symbols, so they are not tried.